### PR TITLE
Include label of whether subject is online or in-person for 2021 subject lists json files

### PR DESCRIPTION
**Pull request by manu354**
Changed `subjectListScraper.ts` to first scrape all online subjects, then scrape all offline subjects, as according to the handbook. `SubjectInfo` now stores a boolean attribute `online` (In addition to `code, name, subjectPeriods`) to note whether the subject is online or in-person. 

Then when the JSON file is written it will contain this new boolean! 

:)

### CODE REVIEW

File: subject-utils/scripts/subjectListScraper.ts [hunk 1]
```diff
@@ -30,11 +33,18 @@ type BaseURL = string;
...
  * @param year The year to search for
  * @param studyPeriod The study period to search for
+ * @param online Whether the subject is online only
  */
-const getBaseURL = (year: number, studyPeriod: SubjectPeriod): BaseURL =>
+const getBaseURL = (year: number, studyPeriod: SubjectPeriod, online:boolean)  =>{
   // tslint:disable-next-line:max-line-length
-  `https://handbook.unimelb.edu.au/search?types[]=subject&year=${year}&subject_level_type[]=all&study_periods[]=${studyPeriod.toLowerCase()}&area_of_study[]=all&org_unit[]=all&campus_and_attendance_mode[]=all&sort=external_code%7Casc`;
-/**
+  if(online){
```
Comment (Trontor) [inline subject-utils/scripts/subjectListScraper.ts:42]: space after close parentheses

File: subject-utils/scripts/subjectListScraper.ts [hunk 2]
```diff
@@ -30,11 +33,18 @@ type BaseURL = string;
...
+    return `https://handbook.unimelb.edu.au/search?types[]=subject&year=${year}&subject_level_type[]=all&study_periods[]=${studyPeriod.toLowerCase()}&area_of_study[]=all&org_unit[]=all&campus_and_attendance_mode[]=Online&sort=external_code%7Casc`;
+  }
```
Comment (Trontor) [inline subject-utils/scripts/subjectListScraper.ts:44]: Code duplication, is there a cleaner way to achieve this?

Maybe pulling out the query parameters out into an array, and constructing the final base URL at once. 
```
const campusAttendanceMode = online ? "Online" ? "all";
const queryParameters = [
'types[]=subject',
`year=${year}`,
 `campus_and_attendance_mode[]=${campusAttendanceMode}` 
 ];
 // Construct final base URL using string utilities (reduce would come in handy here)

```

**Comment by Trontor** [comment]
Instead of two separate runs of the handbook and then a merge -- have you explored any possibilities of parsing online/on campus delivery information from one run?

**Comment by manu354** [comment]
> Instead of two separate runs of the handbook and then a merge -- have you explored any possibilities of parsing online/on campus delivery information from one run?

Yes this was my intial idea, whoever I found no way to make it work, could not find the delivery mode information anywhere in the html.

**Comment by Trontor** [comment]
Neither. Love the idea, I'm assuming this would be useful to flag to the user in some way in a future PR. Would be good to see the updated subject lists output checked into this PR too. You probably already have them on your system.

**Comment by manu354** [comment]
That's the plan! Will do that in the next PR.

Thank you

File: subject-utils/scripts/subjectListScraper.ts (hunk 2 above)
Comment (manu354) [reply subject-utils/scripts/subjectListScraper.ts]: Thanks, I tried to remove the repetition and make it as clean as possible. Let me know if you think the changes might be over-complicating it unnecessarily. 

(check the last commit)

😄

**Comment by manu354** [comment]
I think I have addressed all concerns, let me know if I should incorporate issue #12 into this PR.

**Pull request merged**